tests/s3_data_plane_tests.c

```diff
@@ -8114,3 +8114,52 @@ static int s_test_s3_default_get_without_content_length(struct aws_allocator *al
 
     return AWS_OP_SUCCESS;
 }
+
+struct aws_string *test_helper(struct aws_allocator *allocator) {
+    struct aws_string *host_name =
+        aws_s3_tester_build_endpoint_string(allocator, &g_test_public_bucket_name, &g_test_s3_region);
+    return host_name;
+}
+
+AWS_TEST_CASE(test_s3_default_get, aws_test_s3_default_get)
+static int aws_test_s3_default_get(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    struct aws_s3_tester tester;
+    ASSERT_SUCCESS(aws_s3_tester_init(allocator, &tester));
+
+    struct aws_s3_client *client = NULL;
+    struct aws_s3_tester_client_options client_options;
+    AWS_ZERO_STRUCT(client_options);
+
+    ASSERT_SUCCESS(aws_s3_tester_client_new(&tester, &client_options, &client));
+    struct aws_s3_client_vtable *patched_client_vtable = aws_s3_tester_patch_client_vtable(&tester, client, NULL);
+    patched_client_vtable->http_connection_make_request = s_http_connection_make_request_patch;
+
+    struct aws_s3_meta_request_test_results meta_request_test_results;
+    aws_s3_meta_request_test_results_init(&meta_request_test_results, allocator);
+
+    struct aws_string *hsot_name = test_helper(allocator);
+
+    /* Put together a simple S3 Get Object request. */
```

Comment:
`AWS_ZERO_STRUCT(options)` is used but then only some fields are populated, which can lead to confusion about which fields are intentionally left as zero/NULL.